hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/DatanodeManager.java

```diff
@@ -1829,18 +1829,20 @@ public DatanodeCommand[] handleHeartbeat(DatanodeRegistration nodeReg,
     int totalECBlocks = nodeinfo.getNumberOfBlocksToBeErasureCoded();
     int totalBlocks = totalReplicateBlocks + totalECBlocks;
     if (totalBlocks > 0) {
-      int maxTransfers;
+      int maxECTransfers;
+      int maxReplicationTransfers = blockManager.getMaxReplicationStreams()
+              - xmitsInProgress;;
```

Comment:
There are double semicolons. And I prefer that `int maxECTransfers;` is defined just before the if clause.
```suggestion
      int maxReplicationTransfers = blockManager.getMaxReplicationStreams()
              - xmitsInProgress;
      int maxECTransfers;
```